### PR TITLE
DEVPROD-9426: support writing GitHub app private key to Parameter Store

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-9426

### Description
When writing GitHub app private keys and Parameter Store is enabled for the project, store it in Parameter Store. Unfortunately, because the temporary feature flag is on the project ref, I can't put these functions directly in the `githubapp` package yet or else it'll cause a package dependency cycle. They'll be moved over in DEVPROD-11883 once the project ref feature flag is no longer used.

* Add functions to upsert/remove GitHub app private keys using Parameter Store as the backing storage.
* Switch GitHub app DB calls to use temporary functions for now, until DEVPROD-11883 can move them into the correct package.
* Delete `InsertGitHubAppAuth` because it's unused and didn't seem worth implementing for Parameter Store.

### Testing
* Add unit tests for upsert/remove.
* Tested that the parameter was created/deleted in staging by adding GitHub app auth and then deleting it.

### Documentation
N/A

### CODE REVIEW

File: model/github_app_auth.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	if !isPSEnabled {
+		return
+	}
+
+	op(ref, isRepoRef)
+}
+
+// GitHubAppAuthUpsert upserts the GitHub app auth into the database and to
+// Parameter Store if enabled.
+func GitHubAppAuthUpsert(appAuth *githubapp.GithubAppAuth) error {
```
Comment (hadjri) [inline model/github_app_auth.go:47]: nit: can unexport

File: model/github_app_auth.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+		if err := githubAppAuthRemoveParameterStore(ctx, appAuth); err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":    "could not delete GitHub app auth from Parameter Store",
+				"op":         "Remove",
+				"project_id": appAuth.Id,
+				"epic":       "DEVPROD-5552",
+			}))
+		}
+	}, "Remove")
+	return githubapp.RemoveGithubAppAuth(appAuth.Id)
```
Comment (hadjri) [inline model/github_app_auth.go:112]: Should we skip this DB update if we fail to delete the resource in Parameter store? Same q for upsert

File: model/github_app_auth.go (hunk 1 above)
Comment (Kimchelly) [reply model/github_app_auth.go:47]: Done. I'm not sure it matter really though, since it's temporary code.

File: model/github_app_auth.go (hunk 2 above)
Comment (Kimchelly) [reply model/github_app_auth.go:112]: If we fail to delete it, I think it's okay for now to just continue. If there's a problem with Parameter Store, I wouldn't want that to block the removal of their app auth. The operation will eventually return if it fails to remove the parameter, once we're more confident that the Parameter Store integration is working.

**Review by hadjri — APPROVED** [review]

**Pull request merged**